### PR TITLE
Fix: OOB Invitation against Public DIDs

### PR DESCRIPTION
**Pull request by shaangill025**
- Now able to create and receive OOB invitations against public DIDs.
- Fixes not being able to load `IndyDIDResolver`.
- Implement `supported_did_regex` in `KeyDIDResolver` and `IndyDIDResolver`.

### CODE REVIEW

**Comment by andrewwhitehead** [comment]
We might need to look at where the DID resolver is provided. A BaseLedger instance can't generally be provided globally, because (for indy-sdk at least) it's tied to the wallet. So the DID resolver might need to be provided at the Profile level in that case.  @dbluhm

**Comment by dbluhm** [comment]
I see the issue with the indy ledger loading "fix" I added... @andrewwhitehead the IndyDIDResolver will use the profile passed to the resolve method to get the ledger instance so that is handled correctly, I think. My fix was short-sighted -- the issue we were experiencing was that the ledger disabled setting can be false even while no indy wallet is loaded so the IndyDIDResolver should not be loaded.

File: aries_cloudagent/resolver/__init__.py [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 from ..config.injection_context import InjectionContext
 from ..config.provider import ClassProvider
-from ..ledger.base import BaseLedger
+
+# from ..ledger.base import BaseLedger
```
Comment (andrewwhitehead) [inline aries_cloudagent/resolver/__init__.py:8]: Please remove commented code

File: aries_cloudagent/resolver/default/indy.py [hunk 2]
```diff
@@ -35,9 +39,18 @@ async def setup(self, context: InjectionContext):
...
+        Return supported methods of Indy DID Resolver.
+
+        DEPRECATED: Use supported_did_regex instead.
+        """
         return ["sov"]
 
+    @property
+    def supported_did_regex(self) -> Pattern:
+        """Return supported_did_regex of Indy DID Resolver."""
+        return re.compile(IndyDID.PATTERN)
```
Comment (andrewwhitehead) [inline aries_cloudagent/resolver/default/indy.py:52]: I think it's worth creating this globally and returning a pre-compiled `Pattern` instance.

File: aries_cloudagent/resolver/default/key.py [hunk 3]
```diff
@@ -22,9 +26,18 @@ async def setup(self, context: InjectionContext):
...
+        Return supported methods of Key DID Resolver.
+
+        DEPRECATED: Use supported_did_regex instead.
+        """
         return ["key"]
 
+    @property
+    def supported_did_regex(self) -> Pattern:
+        """Return supported_did_regex of Key DID Resolver."""
+        return re.compile(DIDKeyType.PATTERN)
```
Comment (andrewwhitehead) [inline aries_cloudagent/resolver/default/key.py:39]: Same, can be pre-compiled

Comment (shaangill025) [reply aries_cloudagent/resolver/default/key.py]: I think all requested changes are in.

**Review by andrewwhitehead — APPROVED** [review]

**Pull request merged**